### DISCUSSION TITLE
Calva support for YAMLScript

### DISCUSSION THREAD

**Discussion by ingydotnet** [BetterThanTomorrow/calva#2434] [General]
See: yaml/yamlscript#111 

@PEZ we should schedule a Slack meeting to talk about what this would involve,

**Comment by PEZ** [comment]
Now @ingydotnet and I had had the first of a series of talks about this. We both agree that strong YAMLScript programming support is worth time from both of us. And we have begun to chart out where we have knowledge gaps to fill before making any serious moves.

That said, we are probably looking at a separate extension that would be using Calva's API, which would need to be developed further in directions we're not at all clear about yet.

The next steps here are probably:
* I prepare a presentation around how Calva is built, and how it is using what dependencies. Presenting this to Ingy (and anyone else who may be interested).
* Ingy prepares a similar presentation around YAMLScript to get me more effective in reasoning around the whole endeavor. 

In parallell we need to figure out what we mean by YAMLScript programming support and get a first draft of things to even prioritize between. It reminds me about that Calva started out extremely limited compared to where it is today.

Sorry for all this vagueness. But it reflects where I am with this right now. 😄 Hopefully we can keep updating this discussion and quickly get more clarity.

**Comment by riotrah** [comment]
would be nice to see a public listing of what features either of you had in mind, mvp or beyond.

I'd imagine at a minimum, syntax highlighting and document symbol support?
If this could be done via LSP, there's a lot that could be shared between editors, tho that could grow the scope more than intended if the ambitions are not crazy high to start out.

**Comment by PEZ** [comment]
Yes, we should specify some kind of listing as soon as we can. And we did talk in terms of an LSP for the static stuff.
